apps/web/src/services/dashboards/shared/dashboard-widget-input-form/dashboard-widget-options-form/DashboardWidgetOptionsForm.vue

```diff
@@ -62,12 +76,13 @@ const getMenuHandlers = (schema: WidgetOptionsSchemaProperty): AutocompleteHandl
     if (widgetFormState.inheritOptions?.[schema.key]?.enabled) {
         return [getInheritOptionMenuHandler(schema)];
     }
-    return schema.item_options?.map((conf) => {
-        const options = {}; // e.g. data_source_id: 'ds-1'
+    return schema.item_options?.map((conf: VariableModelConfig) => {
+        const variableModel = _getVariableModel(conf);
+        const options = {}; // e.g. { data_source_id: 'ds-1' }
         Object.entries(schema.dependencies ?? {})?.forEach(([optionName, reference]) => {
             options[reference.reference_key] = widgetFormState.widgetOptions[optionName];
         });
-        return getVariableModelMenuHandler(conf, options);
+        return getVariableModelMenuHandler(variableModel, options);
```

Comment:
suggestion: It doesn't solve the root problem - it must be regenerated every time `widgetFormState.widgetOptions` has been changed.

I think it would be better to be separated like below:
1. add `getVariableModels` which returns `VariableModel[]` by `schema.item_options`.
2. add `variableModelsMap` computed state whose type is `Record<string, VariableModel[]>` by using `getVariableModels()`.
3. change `getMenuHandlers` to return `AutocompleteHandler[]` using `variableModelsMap` state.

Then it will not regenerate variable models every time options have been changed.